cloudinit/config/cc_ssh_import_id.py

```diff
@@ -49,6 +50,13 @@
 
 def handle(_name, cfg, cloud, log, args):
 
+    if cfg.get("ssh_import_id") and not subp.which(SSH_IMPORT_ID_BINARY):
```

Comment:
This is a good catch, but I don't think we can gate it by the compound conditional `cfg.get("ssh_import_id")` because we also have cases like the following that'll trigger the code below and still fail on systems without SSH. 
```yaml
#cloud-config
users:
- name: iwantssh
  ssh_import_id: [falcojr, chad.smith]
```
And even in the len(args) != 0 case below we'd still fail if we don't have SSH_IMPORT_ID_BINARY.  So, we might as well just skip if no SSH_IMPORT_ID_BINARY found.

```suggestion
    if cfg.get("ssh_import_id") and not subp.which(SSH_IMPORT_ID_BINARY):
```

Reply:
Otherwise it might actually be better if we only emit this log when we are trying to import ids via `import_ssh_ids` maybe we just `raise NoSSHImportIDCommandError` and handle that throughout with the single log.warn.

Reply:
@holmanb, per standup discussion, if this PR can handle the above cloud-config wth 'users' with a warning instead of a traceback, that would be great. We can then merge this without any additional broad try/except handling/ or refactor in `import_ssh_ids`.  I think you and James mentioned you'd create a backlog ticket for a more general exception raise/handling for missing package dependendencies for all cloud-init modules. If we have that card logged, we can deal with that separately.